### PR TITLE
feat(route): Penguin Random House

### PR DESCRIPTION
**Pull request by StevenRCE0**
## 完整路由地址 / Example for the proposed route(s)

```routes
/penguin-random-house/the-read-down
/penguin-random-house/articles
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [x] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [x] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

A route to get recommendation lists of books on [Penguin Random House](https://www.penguinrandomhouse.com/the-read-down/).

### CODE REVIEW

**Review by TonyRL — COMMENTED** [review]
It should belong to reading.
Please also add en docs.

File: lib/v2/penguin-random-house/articles.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const utils = require('./utils');
+const cheerio = require('cheerio');
+const got = require('@/utils/got');
+const dateParser = require('@/utils/dateParser');
```
Comment (TonyRL) [inline lib/v2/penguin-random-house/articles.js:4]: Please use parseDate() from parse-date.js

File: lib/v2/penguin-random-house/articles.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+module.exports = async (ctx) => {
+    const link = 'https://www.penguinrandomhouse.com/articles';
+    const res = await got.get(link);
```
Comment (TonyRL) [inline lib/v2/penguin-random-house/articles.js:8]:
```suggestion
    const res = await got(link);
```

**Comment by StevenRCE0** [comment]
Thanks, changed the category and it now passes a Date object to pubDate.

File: lib/v2/penguin-random-house/articles.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+const utils = require('./utils');
+const cheerio = require('cheerio');
+const got = require('@/utils/got');
+
+module.exports = async (ctx) => {
+    const link = 'https://www.penguinrandomhouse.com/articles';
```
Comment (TonyRL) [inline lib/v2/penguin-random-house/articles.js:6]:
```suggestion
    const link = 'https://www.penguinrandomhouse.com/articles/';
```
See https://github.com/DIYgod/RSSHub/runs/7478958103?check_suite_focus=true#step:11:51

File: lib/v2/penguin-random-house/articles.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    const res = await got(link);
+    const $ = cheerio.load(res.data);
+
+    const itemArray = $('.archive-module-half-container,.archive-module-third-container')
+        .map(function () {
+            return {
+                url: $(this).find('a').attr('href'),
+                title: $(this).find('.archive-module-text').first().text(),
+                pubDate: new Date(),
```
Comment (TonyRL) [inline lib/v2/penguin-random-house/articles.js:15]: Don't add a pubDate if there isn't a date.
https://docs.rsshub.app/joinus/pub-date.html

The post date is available from 
image

File: lib/v2/penguin-random-house/router.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+module.exports = function (router) {
+    router.get('/the-read-down', require('./thereaddown'));
+    router.get('/articles', require('./articles'));
```
Comment (TonyRL) [inline lib/v2/penguin-random-house/router.js:3]:
```suggestion
    router.get('/articles', require('./articles'));
    router.get('/the-read-down', require('./thereaddown'));
```
Please sort it in ascending order

File: lib/v2/penguin-random-house/thereaddown.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+const utils = require('./utils');
+const cheerio = require('cheerio');
+const got = require('@/utils/got');
+
+module.exports = async (ctx) => {
+    const link = 'https://www.penguinrandomhouse.com/the-read-down';
```
Comment (TonyRL) [inline lib/v2/penguin-random-house/thereaddown.js:6]:
```suggestion
    const link = 'https://www.penguinrandomhouse.com/the-read-down/';
```

File: lib/v2/penguin-random-house/thereaddown.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    const res = await got(link);
+    const $ = cheerio.load(res.data);
+
+    const itemArray = $('.archive-module-half-container,.archive-module-third-container')
+        .map(function () {
+            return {
+                url: $(this).find('a').attr('href'),
+                title: $(this).find('.archive-module-text').first().text(),
+                pubDate: new Date(),
```
Comment (TonyRL) [inline lib/v2/penguin-random-house/thereaddown.js:15]: Same as above

File: lib/v2/penguin-random-house/utils.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    });
+
+    return {
+        description: descriptionBlock,
+        content: mainBlock,
+    };
+};
+
+const parseList = async (items, ctx, contentParser) =>
+    await Promise.all(
```
Comment (TonyRL) [inline lib/v2/penguin-random-house/utils.js:71]:
```suggestion
    Promise.all(
```

File: lib/v2/penguin-random-house/utils.js [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+        items.map((item) =>
+            ctx.cache.tryGet(item.url, async () => {
+                const itemRes = await got({
+                    method: 'get',
+                    url: item.url,
+                });
```
Comment (TonyRL) [inline lib/v2/penguin-random-house/utils.js:77]:
```suggestion
                const itemRes = await got(item.url);
```
https://github.com/DIYgod/RSSHub/pull/10287#discussion_r928079638

File: lib/v2/penguin-random-house/utils.js [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+
+                const itemPage = itemRes.data;
+                const itemParsed = contentParser(itemPage);
+
+                const result = {
+                    title: item.title,
+                    description: itemParsed.description + '<br>' + itemParsed.content,
+                    pubDate: dateParser(new Date().toISOString()), // No time
```
Comment (TonyRL) [inline lib/v2/penguin-random-house/utils.js:85]: 1. https://github.com/DIYgod/RSSHub/pull/10287#discussion_r928079587
2. Don't add a pubDate if there isn't a date.
https://docs.rsshub.app/joinus/pub-date.html

The post date is available from 
image

**Comment by StevenRCE0** [comment]
Parsed pubDate, I'm using regex for that and it seemed not very tidy...Other mentioned issues are fixed.

File: lib/v2/penguin-random-house/utils.js [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+};
+
+const parseArticle = (element) => {
+    const $ = cheerio.load(element);
+    const description = $('h2.hdr-smalltxt').first().html();
+    const imageSrc = $('div.img-block>img').first().attr('src');
+    const imageAlt = $('div.img-block>img').first().attr('alt');
+    let mainBlock = '';
+
+    const descriptionBlock = art(path.join(__dirname, 'templates/book.art'), {
```
Comment (TonyRL) [inline lib/v2/penguin-random-house/utils.js:75]:
```suggestion
    const descriptionBlock = art(path.join(__dirname, 'templates/articleHeader.art'), {
```
Wrong template used

File: lib/v2/penguin-random-house/utils.js (hunk 8 above)
Comment (LilItalianT82) [reply lib/v2/penguin-random-house/utils.js]: > ```suggestion
>     Promise.all(
> ```

Helping Kelly hood huh?

**Comment by LilItalianT82** [comment]
> Thanks, changed the category and it now passes a Date object to pubDate. 

Helping Kelly hood huh???

**Comment by LilItalianT82** [comment]
Helping Kelly hood huh

**Comment by LilItalianT82** [comment]
Smh Kelly hood

**Pull request merged**